src/test/java/uk/gov/justice/digital/client/glue/GlueSchemaClientTest.java

```diff
@@ -0,0 +1,94 @@
+package uk.gov.justice.digital.client.glue;
+
+import com.amazonaws.SdkClientException;
+import com.amazonaws.services.glue.AWSGlue;
+import com.amazonaws.services.glue.model.EntityNotFoundException;
+import com.amazonaws.services.glue.model.GetSchemaVersionResult;
+import lombok.val;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.justice.digital.config.JobArguments;
+
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class GlueSchemaClientTest {
+
+    private static final String SCHEMA_NAME = "somesource.sometable";
+    private static final String FAKE_SCHEMA_DEFINITION = "This is a fake schema definition";
+
+    @Mock
+    private GlueClientProvider mockClientProvider;
+
+    @Mock
+    private AWSGlue mockClient;
+
+    @Mock
+    private JobArguments mockArguments;
+
+    @Mock
+    private GetSchemaVersionResult mockResponse;
+
+    private GlueSchemaClient underTest;
+
+    @BeforeEach
+    public void setup() {
+        givenJobArgumentsReturnAContractRegistryName();
+        givenClientProviderReturnsAClient();
+        underTest = new GlueSchemaClient(mockClientProvider, mockArguments);
+    }
+
+    @Test
+    public void shouldReturnASchemaForAValidRequest() {
+        givenClientReturnsAValidResponse();
+
+        val result = underTest.getSchema(SCHEMA_NAME);
+
+        assertEquals(Optional.of(FAKE_SCHEMA_DEFINITION), result);
+    }
+
+    @Test
+    public void shouldReturnAnEmptyOptionalForASchemaThatDoesNotExist() {
+        givenClientThrowsAnEntityNotFoundException();
+
+        val result = underTest.getSchema(SCHEMA_NAME);
+
+        assertFalse(result.isPresent());
+    }
+
+    @Test
+    public void shouldThrowAnExceptionIfAnyOtherErrorOccurs() {
+        givenClientThrowsSomeOtherException();
+
+        assertThrows(SdkClientException.class, () -> underTest.getSchema(SCHEMA_NAME));
+    }
+
+    private void givenClientProviderReturnsAClient() {
```

Comment:
this is doing nothing here. remove the test. you are not mocking any behaviour

Reply:
As other remarks - this is setting up mocks for the test.

Reply:
sorry, when doesn't really calling the method. it has no meaning. test it with log statements inside the method and see whether it is getting called. When method itself not called what is the purpose of mocking?

Reply:
The `getClient()` method is being called in line 24 of the GlueSchemaClient. Hence this mock is necessary so that we return a mocked instance of the `AWSGlue` class.